awscli/utils.py

```diff
@@ -277,11 +291,21 @@ def get_pager_stream(self, preferred_pager=None):
 
     @contextlib.contextmanager
     def get_stdout_stream(self):
-        yield get_binary_stdout()
+        yield get_stdout_text_writer()
+
+    def _get_configured_pager(self):
+        return self._session.get_component('config_store').get_config_variable(
+            'pager'
+        )
 
     def _get_process_pager_kwargs(self, pager_cmd):
         kwargs = get_popen_kwargs_for_pager_cmd(pager_cmd)
-        kwargs['stdin'] = subprocess.PIPE
+        kwargs['stdin'] = PIPE
+        env = self._environ.copy()
+        if pager_cmd.startswith('less') and 'LESS' not in env:
```

Comment:
Do you think it makes sense to check for `less` usage? If we set the `LESS` env var and they're not using `less` it doesn't matter right? My concern here is that perhaps someone is using less but the command name is different, such as an alias like `myless` or something.

I tried taking a look at what `git` does (both the source and some testting) and it *seemed* like they will just always set the `LESS` env var if it isn't already set (if the flag was set at build time).

Reply:
If `git` always sets `LESS`, I think that is reasonable, and the alias argument makes sense too.